websites/views.py

```diff
@@ -89,10 +107,105 @@ def get_queryset(self):
         if features.is_enabled(features.USE_LOCAL_STARTERS):
             return WebsiteStarter.objects.all()
         else:
-            return WebsiteStarter.objects.filter(source=STARTER_SOURCE_GITHUB)
+            return WebsiteStarter.objects.filter(source=constants.STARTER_SOURCE_GITHUB)
 
     def get_serializer_class(self):
         if self.action == "list":
             return WebsiteStarterSerializer
         else:
             return WebsiteStarterDetailSerializer
+
+
+class WebsiteCollaboratorViewSet(
+    NestedViewSetMixin,
+    viewsets.ModelViewSet,
+):
+    """ Viewset for Website collaborators along with their group/role """
+
+    serializer_class = WebsiteCollaboratorSerializer
+    permission_classes = (HasWebsiteCollaborationPermission,)
+    pagination_class = DefaultPagination
+    lookup_field = "username"
+
+    def get_queryset(self):
+        """
+        Get a list of all the users with permissions for this website, and annotate by group-name/role
+        (owner, administrator, editor, or global administrator)
+        """
+        website = get_object_or_404(
+            Website, name=self.kwargs.get("parent_lookup_website")
+        )
+        website_groups = list(
+            get_groups_with_perms(website).values_list("name", flat=True)
+        ) + [constants.GLOBAL_ADMIN]
+        owner_username = website.owner.username if website.owner else None
+
+        # Return the individual user and group if a primary key is provided
+        user_name = self.kwargs.get("username", None)
+        if user_name:
+            if user_name == owner_username:
+                return User.objects.filter(username=user_name).annotate(
+                    group=Value(constants.ROLE_OWNER, CharField())
+                )
+            group_subquery = Group.objects.filter(
+                Q(user__username=OuterRef("username")) & Q(name__in=website_groups)
+            )
+            return User.objects.filter(
+                Q(username=user_name) & Q(groups__name__in=website_groups)
+            ).annotate(group=Subquery(group_subquery.values("name")[:1]))
```

Comment:
To me it seems awkward to re-query for the same user we've already fetched for the purpose of adding the annotated group. This group_subquery logic is also kind of difficult to follow. I think we should (a) figure this out in the view and pass it into the serializer via context, or (b) let the serializer itself figure it out. If we go with (b) for eample, we could pass `owner_username` to the serializer via context, and add `get_group` to the serializer. If the `group` attribute exists on the object, just return that. If not, run this logic to compare the owner username, then fetch the relevant group name separately if the username doesn't match. That method could probably be cached as well so we avoid doubling queries when serializing the `group` field and using the group value in `get_role`. This adds some responsibilities to the serializer but I think it's worth it.

Once we move that logic, we can get rid of the multiple `Group.objects.get` queries. You could just fetch the records in their entirety right away instead of just fetching the `name`.